File: qb-engineer-server/qb-engineer.core/Models/ShipmentListItemModel.cs
namespace QBEngineer.Core.Models;

public record ShipmentListItemModel(
    int Id,
    string ShipmentNumber,
    int SalesOrderId,
    string SalesOrderNumber,
    string CustomerName,
    string Status,
    string? Carrier,
    string? TrackingNumber,
    DateTimeOffset? ShippedDate,
    DateTimeOffset CreatedAt);

File: qb-engineer-ui/src/app/features/shipments/models/shipment-list-item.model.ts
export interface ShipmentListItem {
  id: number;
  shipmentNumber: string;
  salesOrderId: number;
  salesOrderNumber: string;
  customerName: string;
  status: string;
  carrier: string | null;
  trackingNumber: string | null;
  shippedDate: Date | null;
  createdAt: Date;
}
